src/structure.cpp

```diff
@@ -216,6 +216,31 @@ void structure::choose_chunkdivision(const grid_volume &thegv, int desired_num_c
   check_chunks();
 }
 
+double structure::chunk_cost(int i) const {
+  return i < 0 || i > num_chunks ? 0.0 : chunks[i]->gv.get_cost();
+}
+
+void structure::print_estimated_costs() const {
+  double *costs = new double[count_processors()];
+  for (int i = 0; i < count_processors(); i++)
+    costs[i] = 0;
+  for (int i = 0; i < num_chunks; i++)
+    costs[chunks[i]->n_proc()] += chunk_cost(i);
```

Comment:
For some reason, this line seems to be producing a segmentation fault at runtime even though `make` for this branch/PR finishes without errors.
```
[simpetus:16699] [ 0] /lib/x86_64-linux-gnu/libpthread.so.0(+0x12890)[0x7f85427c3890]
[simpetus:16699] [ 1] /usr/local/lib/libctlgeom.so.7(+0x3bc4)[0x7f8540759bc4]
[simpetus:16699] [ 2] /usr/local/lib/libctlgeom.so.7(geom_fix_object_ptr+0x479)[0x7f8540761549]
[simpetus:16699] [ 3] /usr/local/lib/libctlgeom.so.7(geom_get_bounding_box+0x34)[0x7f8540761da4]
[simpetus:16699] [ 4] /usr/local/lib/libctlgeom.so.7(overlap_with_object+0x123)[0x7f8540763103]
[simpetus:16699] [ 5] /home/oskooi/install/meep6/src/.libs/libmeep.so.16(_ZN9meep_geom14fragment_stats13compute_statsEv+0x9a)[0x7f8540c156aa]
[simpetus:16699] [ 6] /home/oskooi/install/meep6/src/.libs/libmeep.so.16(_ZN9meep_geom14fragment_stats7computeEv+0x9)[0x7f8540c16269]
[simpetus:16699] [ 7] /home/oskooi/install/meep6/src/.libs/libmeep.so.16(_ZNK4meep11grid_volume8get_costEv+0x55)[0x7f8540be5555]
[simpetus:16699] [ 8] /home/oskooi/install/meep6/src/.libs/libmeep.so.16(_ZNK4meep9structure21print_estimated_costsEv+0xc3)[0x7f8540bab773]
[simpetus:16699] [ 9] /home/oskooi/install/meep6/python/meep/_meep.so(+0x62533)[0x7f8540ebc533]
[simpetus:16699] [10] corrupted double-linked list
```

Reply:
Maybe add a `master_printf("n_proc[%d] = %d\n", i, chunks[i]->n_proc());` line above this line to see what the value of `n_proc` is?

Reply:
`n_proc()` is not the issue (i.e., output is as expected), rather it's the function call `chunks[i]->gv.get_cost()` inside `chunk_cost(int i)` which is causing the error. Could this have something to do with [`get_cost()`](https://github.com/NanoComp/meep/blob/master/src/vec.cpp#L1005-L1010) being a `const` function?